exercises/change/src/Example.hs

```diff
@@ -0,0 +1,29 @@
+module Change (findFewestCoins) where
+
+import Data.List (sortBy)
+
+type Coin = Integer
+type Coins = [Coin]
+type Amount = Integer
+
+findFewestCoins :: Amount -> Coins -> Maybe Coins
+findFewestCoins target coins = minChange target sortedCoins [] Nothing
+  where
+    sortedCoins = sortBy (flip compare) coins
+
+    minChange target coins candidate bestResult
+      | target < 0 || worseResult = bestResult
+      | target == 0 = Just candidate
+      | otherwise = dropCoin target coins candidate newBestResult
+      where
+        worseResult = maybe False (\x -> length x <= length candidate) bestResult
+        newBestResult = addCoin target coins candidate bestResult
+
+    addCoin target coins@(coin:_) candidate bestResult
+      | newTarget >= 0 = minChange newTarget coins (coin:candidate) bestResult
+      where newTarget = target - coin
+    addCoin _ _ _ bestResult = bestResult
+
+    dropCoin target (_:restCoins) candidate bestResult =
```

Comment:
In case you are having problems reproducing the error in Travis, try to test your solution with:

```
stack clean; stack test --pedantic
```

If it passes this test, it will probably work in Travis.